src/router/PrivateRoute.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Route, Redirect } from 'react-router-dom';
+
+import Header from '../components/Header';
+import { LOGIN_PAGE_ROUTE } from '../constants/routes';
+
+export const PrivateRoute = ({ isAuthenticated, component: Component, ...rest }) => (
+  <Route
+    {...rest}
+    component={props => {
```

Comment:
you are passing a function, so you should use `render` prop. `component` and `render` has differences. 
also:
`props => isAuthenticated ? <><Header />.... </> : <Redirect>` 
I am not sure about this approach at all, but at this moment let's stay as is if works correctly